frontend/src/components/sign-up/sign-up.tsx

```diff
@@ -1,99 +1,109 @@
+import { useForm, FieldValues } from 'react-hook-form';
```

Comment:
since we'll use this markup for both pages (sign-in, sign-up), lets remove this component to auth:

```
frontend/src/components/auth/auth.ts
frontend/src/components/components/sign-up-form/sign-up-form.tsx // your sign-up form
frontend/src/components/components/sign-in-form/sign-in-form.tsx // sign-in form that will be in the future
```

then, in the `frontend/src/components/auth/auth.ts`, we can do such a thing:

```
const Auth = () => {
  const { pathname } = useLocation();

  const getScreen = (screen: AppRoute) => {
    switch (screen) {
      case AppRoute.SIGN_UP: {
        <SignUpForm onSubmit={handleSignUpSubmit} />
      }
      case AppRoute.SIGN_IN: {
        <SignInForm onSubmit={handleSignInSubmit} />
      }
    }

    return null;
  };

 return <>
  {getScreen(pathname)}
  </>
};
```